honeybee_grasshopper_radiance/src/HB Extract HDR.py

```diff
@@ -0,0 +1,218 @@
+            view.
+        _resolution_: An integer for the dimension of the output image in pixels.
+            If extracting a 180 FOV angular or hemispherical HDR image from a
+            360 FOV HDR image, the default resolution is 1/3 of the resolution of
+            _hdr. If converting between 180 FOV angular or hemispherical HDR 
+            images, the default resolution is that of _hdr.
+        _hdr: Path to a High Dynamic Range (HDR) image file from which to
+            interpolate or extrapolate a view.
+
+    Returns:
+        hdr: Path to the resulting HDR image file.
+"""
+
+ghenv.Component.Name = 'HB Extract HDR'
+ghenv.Component.NickName = 'ExtractHDR'
+ghenv.Component.Message = '1.3.0'
+ghenv.Component.Category = 'HB-Radiance'
+ghenv.Component.SubCategory = '4 :: Results'
+ghenv.Component.AdditionalHelpFromDocStrings = '3'
+
+import os
+import subprocess
+
+try:  # import honeybee_radiance_command dependencies
+    from honeybee_radiance_command.pinterp import Pinterp
+    from honeybee_radiance_command.ra_xyze import Ra_xyze
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_radiance_command:\n\t{}'.format(e))
+
+try:  # import honeybee_radiance dependencies
+    from honeybee_radiance.config import folders as rad_folders
+    from honeybee_radiance.view import View
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_radiance:\n\t{}'.format(e))
+
+try:  # import ladybug_rhino dependencies
+    from ladybug_rhino.grasshopper import all_required_inputs, give_warning
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+# check the Radiance date of the installed radiance
+try:  # import lbt_recipes dependencies
+    from lbt_recipes.version import check_radiance_date
+except ImportError as e:
+    raise ImportError('\nFailed to import lbt_recipes:\n\t{}'.format(e))
+check_radiance_date()
+
+
+def check_view_hdr(hdr_path):
+    """Check if the header of the HDR image contains a view (VIEW=).
+    
+    A ValueError is raised if the image does not contain a valid view.
+    A ValueError is raised if the view type is not -vta or -vth.
+    
+    Args:
+        hdr_path: The path to an HDR image file.
+    """
+    # set valid view = False. Will later be set to True if a valid view is found
+    valid_view = False
+    
+    # read hdr image and search for a valid view
+    with open(hdr_path, 'r') as hdr_file:
+        for lineCount, line in enumerate(hdr_file):
+            if lineCount < 200:
+                low_line = line.lower()
+                if not low_line.startswith('\t'):
+                    if low_line.startswith('view='):
+                        valid_view = True
+                        if line.find('-vth') > -1:
+                            projection = '-vth'
+                        elif line.find('-vta') > -1:
+                            projection = '-vta'
```

Comment:
We don't need these two checks here if you are just going to check the view type on the object later.